config/main.py

```diff
@@ -823,6 +822,10 @@ def startup(ctx):
     config_db = ctx.obj['config_db']
     interface_name = ctx.obj['interface_name']
 
+    alias = interface_name_to_alias(interface_name)
+    if alias is None:
```

Comment:
Interface alias is currently an optional attribute in SONiC. Since it is not mandatory, it is valid for a platform to not define any aliases. On those platforms, this check will always fail.

Therefore, it would be be better test to check if the interface name exists in the port table, rather than checking for the existence of an alias.

Reply:
I agree with your view. 'alias' is an optional attribute. Since most of the other functions in main.py is using interface_name_to_alias() routine for checking the presence of interface, i reused it.
But like you said, i agree we should have a better check about the interface name in the PORT table in config db.
I will do the changes.

Reply:
Joe, the vstest check failed for the PR build after resubmit of the fixes.
But from the console logs, i see nothing related to the fixes in this PR that could have resulted in the build failure. How do I proceed from here? Appreciate your inputs. Thanks.

Reply:
The vstest check is currently broken. Don't worry about it.